### DISCUSSION TITLE
React PHP API Throttling

### DISCUSSION THREAD

**Discussion by arunnabraham** [reactphp/reactphp#498] [General]
I would like to know whether we able to do easy API throttling? I saw the doc says that could be done with MQ. 

https://github.com/clue/reactphp-mq. 

So is there a way that we can hard limit the api calls to a defined number? Let me know if we can build a mechanism for let say like 4 apis to call at once. but throttling would be like 20 req/sec

**Comment by SimonFrings** [comment]
@arunnabraham thanks for bringing this up :+1:

There already have been a few discussions about this in https://github.com/clue/reactphp-mq/issues/14, https://github.com/clue/reactphp-flux/issues/17 and https://github.com/clue/reactphp-mq/issues/26. I think @clue's quote here brings it straight to the point:
> This is indeed possible using this project's architecture and I've indeed used this in a number of client projects already. I'd love to work out a reusable API once we have a better understanding of how much we can actually reuse in here, but so far this always turned out to be a rather individual integration. If you need this for a commercial project and you want to help sponsor this feature, feel free to reach out and I'm happy to take a look.

Any ideas on how we can achieve a reusable implementation for this are welcome!
